src/Model/SearchTrait.php

```diff
@@ -129,7 +141,7 @@ protected function _extractParams($params, $filters)
      * ```
      *
      * @param array $params The parameters array to flatten.
-     * @param array $filters The array of filters with configuration
+     * @param \Search\Model\Filter\Base[] $filters The array of filters with configuration
      * @return array The flattened parameters array.
      */
     protected function _flattenParams($params, $filters)
```

Comment:
Ideally we also typehint the array on such a protected methods to make sure they are of that type going in.